### PR TITLE
Add sidebar navigation

### LINKED ISSUE #1: Table of contents for CV
**Is your feature request related to a problem? Please describe.**
When I'm looking at the CV I don't know what sections are present.

**Describe the solution you'd like**
Something similar to devscard with the section icons on the right bar
image

**Describe alternatives you've considered**
Menu on the top.

**Additional context**

### PR DESCRIPTION
**Pull request by KonradSzwarc**
Closes #1

### CODE REVIEW

File: src/pages/index.astro [hunk 1]
```diff
@@ -43,8 +44,18 @@ await initializeWebContext(Astro, {
...
-  <Section class="gap-8">
+  <Sidebar>
+    <SidebarItem sectionId="basics" icon="fa6-solid:user">About me</SidebarItem>
+    <SidebarItem sectionId="skills" icon="fa6-solid:bars-progress">Skills</SidebarItem>
+    <SidebarItem sectionId="jobs" icon="fa6-solid:suitcase">Work experience</SidebarItem>
+    <SidebarItem sectionId="projects" icon="fa6-solid:rocket">Projects</SidebarItem>
+    <SidebarItem sectionId="references" icon="fa6-solid:comment">References</SidebarItem>
+    <SidebarItem sectionId="achievements" icon="fa6-solid:trophy">Achievements</SidebarItem>
+    <SidebarItem sectionId="education" icon="fa6-solid:graduation-cap">Education</SidebarItem>
+    <SidebarItem sectionId="favorites" icon="fa6-solid:star">Favorites</SidebarItem>
```
Comment (MateuszNaKodach) [inline src/pages/index.astro:55]: Może tak miało być, ale brakuje Interests na side barze.

**Review by MateuszNaKodach — APPROVED** [review]

File: src/pages/index.astro [hunk 2]
```diff
@@ -43,8 +44,18 @@ await initializeWebContext(Astro, {
     <ElementColors />
     <Metadata entry={getEntry('metadata', 'main')} />
   </Fragment>
-  <Basics entry={getEntry('basics', 'main')} />
-  <Section class="gap-8">
+  <Sidebar>
+    <SidebarItem sectionId="basics" icon="fa6-solid:user">About me</SidebarItem>
```
Comment (MateuszNaKodach) [inline src/pages/index.astro:48]: It looks strange, when you scroll to basics, I'd expect to scroll from the top, but looks like that:

image

I understand why, but would be better to scroll to the top of the page like that:
image

File: src/pages/index.astro (hunk 1 above)
Comment (KonradSzwarc) [reply src/pages/index.astro:55]: It's intentional not to put interests into the sidebar. It showcases you can skip some sections in the sidebar if you consider them less important.

File: src/pages/index.astro (hunk 2 above)
Comment (KonradSzwarc) [reply src/pages/index.astro:48]: Scrolling to the top of the page may be difficult. There is no element to scroll to, and (as visible on the second screenshot) when you're at the top, no section is selected on the sidebar. However, I've added scroll margins to all sections so you can test it again.

Comment (MateuszNaKodach) [reply src/pages/index.astro:48]: Thanks for the explanation.
Now looks better than before :)

**Pull request merged**